libs/ngx-sherfire/src/lib/firebase-config.service.ts

```diff
@@ -1,15 +1,9 @@
-import type firebase from 'firebase/app';
+import type { FirebaseAppSettings, FirebaseOptions } from 'firebase/app';
+import type { FirestoreSettings } from 'firebase/firestore';
 
 export abstract class FirebaseConfig {
-    abstract apiKey: string;
```

Comment:
There is no need to have all these abstract fields now that we augment the FirebaseConfig type with FirebaseOptions (see new line 9)

Reply:
But are they now intentionally optional?

Reply:
Yes, I don't really know why we should make (some) fields mandatory. It all depends on which firebase features you are going to use.